tests/server/test_base_server.py

```diff
@@ -126,7 +126,7 @@ def test_metrics_reporter_fit_round(mock_fit_round: Mock) -> None:
     }
 
 
-@patch("fl4health.server.base_server.Server.evaluate_round")
+@patch("fl4health.server.base_server.FlServer._evaluate_round")
 @freeze_time("2012-12-12 12:12:12")
 def test_metrics_reporter_evaluate_round(mock_evaluate_round: Mock) -> None:
     test_round = 2
```

Comment:
I think there are two methods that we want to test in the base server that currently aren't. The first is the functionality of `_unpack_metrics` and the second is `_handle_result_aggregation` from the suggestions above. You should be able to mock up all of the relevant pieces to test those methods out thoroughly using the basic fed avg strategy and our default aggregation schemes in `from fl4health.utils.metric_aggregation`

Reply:
added the tests in the latest commit!